v1/entitlement/entitlement.gunk

```diff
@@ -0,0 +1,583 @@
+//                                                 "write": "Grants write access",
+//                                         },
+//                                 },
+//                         },
+//                 },
+//         },
+// }
+package entitlement
+
+import (
+	"time"
+
+	"github.com/gunk/opt/http"
+	"github.com/gunk/opt/openapiv2"
+)
+
+type AddEntitlementRequestForCurrentUserRequest struct {
+	BankID   string `pb:"1" json:"bank_id"`
+	RoleName string `pb:"2" json:"role_name"`
+}
+
+type EntitlementRequest struct {
+	EntitlementRequestID string    `pb:"1" json:"entitlement_request_id"`
+	User                 User      `pb:"2" json:"user"`
+	RoleName             string    `pb:"3" json:"role_name"`
+	BankID               string    `pb:"4" json:"bank_id"`
+	Created              time.Time `pb:"5" json:"created"`
+}
+
+type EntitlementRequests struct {
+	EntitlementRequests []EntitlementRequest `pb:"1" json:"entitlement_requests"`
+}
+
+type List struct {
+	EntitlementID string `pb:"1" json:"entitlement_id"`
+	RoleName      string `pb:"2" json:"role_name"`
+	BankID        string `pb:"3" json:"bank_id"`
+}
+
+type Entitlements struct {
+	List []List `pb:"1" json:"list"`
+}
+
+type User struct {
+	UserID       string       `pb:"1" json:"user_id"`
+	Email        string       `pb:"2" json:"email"`
+	ProviderID   string       `pb:"3" json:"provider_id"`
+	Provider     string       `pb:"4" json:"provider"`
+	Username     string       `pb:"5" json:"username"`
+	Entitlements Entitlements `pb:"6" json:"entitlements"`
+}
+
+type AddEntitlementRequestForUserRequest struct {
+	UserID   string `pb:"1" json:"user_id"`
+	BankID   string `pb:"2" json:"bank_id"`
+	RoleName string `pb:"3" json:"role_name"`
+}
+
+type AddEntitlementRequestForUserResponse struct {
+	UserID        string `pb:"1" json:"user_id"`
+	EntitlementID string `pb:"2" json:"entitlement_id"`
+	RoleName      string `pb:"3" json:"role_name"`
+	BankID        string `pb:"4" json:"bank_id"`
+}
+
+type DeleteEntitlementRequest struct {
+	UserID        string `pb:"1" json:"user_id"`
+	EntitlementID string `pb:"2" json:"entitlement_id"`
+}
+
+type DeleteEntitlementRequestRequest struct {
+	EntitlementRequestID string `pb:"1" json:"entitlement_request_id"`
+}
+
+type GetEntitlementRequestForUserRequest struct {
+	UserID string `pb:"1" json:"user_id"`
+}
+
+type GetEntitlementRequestForUserResponse struct {
+	EntitlementRequests []EntitlementRequest `pb:"1" json:"entitlement_requests"`
+}
+
+type GetEntitlementRequestForCurrentUserRequest struct {
+}
+
+type GetEntitlementRequestForCurrentUserResponse struct {
+	EntitlementRequests []EntitlementRequest `pb:"1" json:"entitlement_requests"`
+}
+
+type GetEntitlementForUserAtBankRequest struct {
+	BankID string `pb:"1" json:"bank_id"`
+	UserID string `pb:"2" json:"user_id"`
+}
+
+type GetEntitlementForUserRequest struct {
+	UserID string `pb:"2" json:"user_id"`
+}
+
+type GetEntitlementForCurrentUserRequest struct {
```

Comment:
this one also

Reply:
Because there's nothing to pass, but it may be used in the future to add arguments if any. Honestly, I followed https://github.com/openbank/openbank/blob/master/v1/customer/customer.gunk#L302